### ISSUE TITLE
Error when run "bash extract_frames.sh" 

### ISSUE THREAD

**Issue by cediy2088** [open-mmlab/mmaction#204]
```
(py37env) shenl@robot-deep-learning:~/mmaction/mmaction/data_tools/ucf101$ bash extract_frames.sh |more
Reading videos from folder:  ../data/ucf101/videos/
Extension of videos:  avi
Total number of videos found:  13320
**VIDIOC_REQBUFS: Inappropriate ioctl for device**
2020-08-06 22:31:40,148 FATAL [default] Check failed: [video_stream.isOpened()] Cannot open video stream "../data/ucf101/videos/SkyDiving/v_SkyDiving_g02_c02.avi" for optical flow extraction.
2020-08-06 22:31:40,148 WARN  [default] Aborting application. Reason: Fatal log at [/home/shenlong/mmaction/mmaction/third_party/dense_flow/src/dense_flow_gpu.cpp:19]
VIDIOC_REQBUFS: Inappropriate ioctl for device
2020-08-06 22:31:40,154 FATAL [default] Check failed: [video_stream.isOpened()] Cannot open video stream "../data/ucf101/videos/SkateBoarding/v_SkateBoarding_g20_c02.avi" for optical flow extraction.
2020-08-06 22:31:40,154 WARN  [default] Aborting application. Reason: Fatal log at [/home/shenlong/mmaction/mmaction/third_party/dense_flow/src/dense_flow_gpu.cpp:19]
VIDIOC_REQBUFS: Inappropriate ioctl for device
2020-08-06 22:31:40,157 FATAL [default] Check failed: [video_stream.isOpened()] Cannot open video stream "../data/ucf101/videos/Mixing/v_Mixing_g05_c05.avi" for optical flow extraction.
2020-08-06 22:31:40,158 WARN  [default] Aborting application. Reason: Fatal log at [/home/shenlong/mmaction/mmaction/third_party/dense_flow/src/dense_flow_gpu.cpp:19]
VIDIOC_REQBUFS: Inappropriate ioctl for device
2020-08-06 22:31:40,162 FATAL [default] Check failed: [video_stream.isOpened()] Cannot open video stream "../data/ucf101/videos/Nunchucks/v_Nunchucks_g17_c04.avi" for optical flow extraction.
2020-08-06 22:31:40,162 WARN  [default] Aborting application. Reason: Fatal log at [/home/shenlong/mmaction/mmaction/third_party/dense_flow/src/dense_flow_gpu.cpp:19]
VIDIOC_REQBUFS: Inappropriate ioctl for device
2020-08-06 22:31:40,163 FATAL [default] Check failed: [video_stream.isOpened()] Cannot open video stream "../data/ucf101/videos/ShavingBeard/v_ShavingBeard_g24_c02.avi" for optical flow extraction.
2020-08-06 22:31:40,164 WARN  [default] Aborting application. Reason: Fatal log at [/home/shenlong/mmaction/mmaction/third_party/dense_flow/src/dense_flow_gpu.cpp:19]
VIDIOC_REQBUFS: Inappropriate ioctl for device
2020-08-06 22:31:40,167 FATAL [default] Check failed: [video_stream.isOpened()] Cannot open video stream "../data/ucf101/videos/SumoWrestling/v_SumoWrestling_g07_c02.avi" for optical flow extraction.
2020-08-06 22:31:40,167 WARN  [default] Aborting application. Reason: Fatal log at [/home/shenlong/mmaction/mmaction/third_party/dense_flow/src/dense_flow_gpu.cpp:19]
VIDIOC_REQBUFS: Inappropriate ioctl for device
2020-08-06 22:31:40,171 FATAL [default] Check failed: [video_stream.isOpened()] Cannot open video stream "../data/ucf101/videos/BalanceBeam/v_BalanceBeam_g23_c05.avi" for optical flow extraction.
2020-08-06 22:31:40,171 WARN  [default] Aborting application. Reason: Fatal log at [/home/shenlong/mmaction/mmaction/third_party/dense_flow/src/dense_flow_gpu.cpp:19]
VIDIOC_REQBUFS: Inappropriate ioctl for device
2020-08-06 22:31:40,173 FATAL [default] Check failed: [video_stream.isOpened()] Cannot open video stream "../data/ucf101/videos/LongJump/v_LongJump_g09_c01.avi" for optical flow extraction.
2020-08-06 22:31:40,173 WARN  [default] Aborting application. Reason: Fatal log at [/home/shenlong/mmaction/mmaction/third_party/dense_flow/src/dense_flow_gpu.cpp:19]
Aborted (core dumped)
cmd../../mmaction/third_party/dense_flow/build/extract_gpu -f=../data/ucf101/videos/SkyDiving/v_SkyDiving_g02_c02.avi -x=../data/ucf101/rawframes/SkyDiving/v_SkyDiving_g02_c02/flow_x -y=../data/ucf101/rawframes/SkyDiving/v_SkyDiving_g02_c02/flow_y -i=../data/ucf101/rawfr
ames/SkyDiving/v_SkyDiving_g02_c02/img -b=20 -t=1 -d=1 -s=1 -o=dir -w=0 -h=0
417 SkyDiving/v_SkyDiving_g02_c02 done
Aborted (core dumped)
cmd../../mmaction/third_party/dense_flow/build/extract_gpu -f=../data/ucf101/videos/SumoWrestling/v_SumoWrestling_g07_c02.avi -x=../data/ucf101/rawframes/SumoWrestling/v_SumoWrestling_g07_c02/flow_x -y=../data/ucf101/rawframes/SumoWrestling/v_SumoWrestling_g07_c02/flow_y
 -i=../data/ucf101/rawframes/SumoWrestling/v_SumoWrestling_g07_c02/img -b=20 -t=1 -d=4 -s=1 -o=dir -w=0 -h=0
1668 SumoWrestling/v_SumoWrestling_g07_c02 done
Aborted (core dumped)
cmd../../mmaction/third_party/dense_flow/build/extract_gpu -f=../data/ucf101/videos/Mixing/v_Mixing_g05_c05.avi -x=../data/ucf101/rawframes/Mixing/v_Mixing_g05_c05/flow_x -y=../data/ucf101/rawframes/Mixing/v_Mixing_g05_c05/flow_y -i=../data/ucf101/rawframes/Mixing/v_Mixi
ng_g05_c05/img -b=20 -t=1 -d=7 -s=1 -o=dir -w=0 -h=0
2919 Mixing/v_Mixing_g05_c05 done
Aborted (core dumped)
cmd../../mmaction/third_party/dense_flow/build/extract_gpu -f=../data/ucf101/videos/Nunchucks/v_Nunchucks_g17_c04.avi -x=../data/ucf101/rawframes/Nunchucks/v_Nunchucks_g17_c04/flow_x -y=../data/ucf101/rawframes/Nunchucks/v_Nunchucks_g17_c04/flow_y -i=../data/ucf101/rawfr
ames/Nunchucks/v_Nunchucks_g17_c04/img -b=20 -t=1 -d=0 -s=1 -o=dir -w=0 -h=0
0 Nunchucks/v_Nunchucks_g17_c04 done
Aborted (core dumped)
cmd../../mmaction/third_party/dense_flow/build/extract_gpu -f=../data/ucf101/videos/BalanceBeam/v_BalanceBeam_g23_c05.avi -x=../data/ucf101/rawframes/BalanceBeam/v_BalanceBeam_g23_c05/flow_x -y=../data/ucf101/rawframes/BalanceBeam/v_BalanceBeam_g23_c05/flow_y -i=../data/
ucf101/rawframes/BalanceBeam/v_BalanceBeam_g23_c05/img -b=20 -t=1 -d=2 -s=1 -o=dir -w=0 -h=0
834 BalanceBeam/v_BalanceBeam_g23_c05 done
Aborted (core dumped)
cmd../../mmaction/third_party/dense_flow/build/extract_gpu -f=../data/ucf101/videos/SkateBoarding/v_SkateBoarding_g20_c02.avi -x=../data/ucf101/rawframes/SkateBoarding/v_SkateBoarding_g20_c02/flow_x -y=../data/ucf101/rawframes/SkateBoarding/v_SkateBoarding_g20_c02/flow_y
 -i=../data/ucf101/rawframes/SkateBoarding/v_SkateBoarding_g20_c02/img -b=20 -t=1 -d=5 -s=1 -o=dir -w=0 -h=0
2085 SkateBoarding/v_SkateBoarding_g20_c02 done
Aborted (core dumped)
cmd../../mmaction/third_party/dense_flow/build/extract_gpu -f=../data/ucf101/videos/ShavingBeard/v_ShavingBeard_g24_c02.avi -x=../data/ucf101/rawframes/ShavingBeard/v_ShavingBeard_g24_c02/flow_x -y=../data/ucf101/rawframes/ShavingBeard/v_ShavingBeard_g24_c02/flow_y -i=..
/data/ucf101/rawframes/ShavingBeard/v_ShavingBeard_g24_c02/img -b=20 -t=1 -d=3 -s=1 -o=dir -w=0 -h=0
1251 ShavingBeard/v_ShavingBeard_g24_c02 done
VIDIOC_REQBUFS: Inappropriate ioctl for device

```
the videos path is ok，Who can tell me what is wrong?

**Comment by zhaoyue-zephyrus** [comment]
Hi @cediy2088 
According to [this question](https://stackoverflow.com/questions/41200201/opencv-unable-to-stop-the-stream-inappropriate-ioctl-for-device#:~:text=The%20problem%20was%20due%20to%20missing%20ffmpeg%20and%20ffmpeg%2Ddevel.&text=And%20if%20you%20compile%20and,and%20%22make%20install%22%20again.) You may need to reinstall your opencv with `-DWITH_LIBV4L=ON` particularly.

**Comment by cediy2088** [comment]
> Hi @cediy2088
> According to [this question](https://stackoverflow.com/questions/41200201/opencv-unable-to-stop-the-stream-inappropriate-ioctl-for-device#:~:text=The%20problem%20was%20due%20to%20missing%20ffmpeg%20and%20ffmpeg%2Ddevel.&text=And%20if%20you%20compile%20and,and%20%22make%20install%22%20again.) You may need to reinstall your opencv with `-DWITH_LIBV4L=ON` particularly.

I have solved the problom, recompile opencv-4.1.0 with -DWITH_FFMPEG=ON
